test/controllers/users_controller_test.rb

```diff
@@ -32,4 +32,41 @@
       expect(flash[:result_text]).must_equal "User not found!"
     end
   end
+
+  describe "login" do
+    it "can log in an existing user" do
+      # Arrange
+      user_count = User.count
+
+      # Act
+      user = perform_login
+
+      expect(user_count).must_equal User.count
+
+      # Should also test Flash notices
```

Comment:
absolutely agree! i might have a test that says: expect(flash[:success]).must_equal "Logged in as returning user #{user.username}"